src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -2086,29 +2079,6 @@ private final class MapKeyValueTupleReverseIterator[K, V](rootNode: MapNode[K, V
   }
 }
 
-private final class MapKeyValueTupleHashIterator[K, V](rootNode: MapNode[K, V])
```

Comment:
I thought that for a HashMap, the hashcode was optimised to not recalculate the hash of each key, as it was stable. Doesn't this change remove that optimisation

Reply:
Good point! I've restored this class but removed a bit of indirection to directly call `tuple2Hash` rather than implementing `Product2`.